File: 875-koko-eating-bananas/875-koko-eating-bananas.cpp
class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int left = 1, right = *max_element(piles.begin(), piles.end());
        while(left != right) {
            int mid = (left+right)/2;
            int hours = 0;
            for(auto pile: piles) {
                hours += pile / mid + (pile % mid != 0);
            }
            if(hours <= h) {
                right = mid;
            }
            else {
                left = mid+1;
            }
        }
        return left;
    }
};

File: 875-koko-eating-bananas/875-koko-eating-bananas.py
from math import ceil
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        left = 1
        right = max(piles)
        
        while left != right:
            hours = 0
            mid = (left+right)//2
            for pile in piles:
                hours += math.ceil(pile/mid)
            if hours <= h:
                right = mid
            else:
                left = mid+1
        return left